### ISSUE TITLE
Add more diagnostic logging in MVC

### ISSUE THREAD

**Issue by rynowak** [aspnet/Mvc#3193]
From offline logging discussion:

Add Verbose logging where it can be useful for debugging
- Time box to 2 dev-days maximum
- Use EventIds at each logging statement
- Update any existing logging to include an EventId

Relevant logging points we want to include for sure:

```
MVC:
        - Controller/Action (about to be run, would like to include MB data, include ModelState.IsValid) - Information
        - Action Result (each of them) - Information

```

**Comment by Eilon** [comment]
@ryanbrandenburg please turn 180 degrees and find the nearest other Ryan. He can point you in the right direction for this :smile: 

This PR should end up fairly similar to https://github.com/aspnet/Security/pull/530

**Comment by rynowak** [comment]
I was hoping we could also incorporate the new logging paradigm here.

**Comment by Eilon** [comment]
Ah using LoggerDefinitions? That would be awesome!

**Comment by rynowak** [comment]
da731fc641f56529328d3cb74d6092310a40627e contains part of this, there is still plenty of logging using the older paradigm that needs to be updated

**Comment by ryanbrandenburg** [comment]
Fixed in #3387
